chain-test/src/test/scala/org/bitcoins/chain/util/ChainFixture.scala

```diff
@@ -0,0 +1,27 @@
+package org.bitcoins.chain.util
+
+import org.bitcoins.chain.blockchain.ChainHandler
+import org.bitcoins.chain.models.BlockHeaderDAO
+
+/**
+  * This ADT represents all Chain test fixtures. If you set this type to be your
+  * FixtureParam and override withFixture to be withChainFixutre, then simply tag
+  * tests to specify which fixture that test should receive and then use inFixutred
+  * which takes a PartialFunction[ChainFixture, Future[Assertion] ] (i.e. just
+  * specify the relevant case for your expected fixture)
```

Comment:
Nit: some markdown syntax works in Scaladoc. So I'd write \`withFixture\`, \`PartialFunction[ChainFixture, Future[Assertion]]\` etc. Makes it easier for the lib user to quickly parse the doc:-)